src/test/java/com/redislabs/redisgraph/RedisGraphAPITest.java

```diff
@@ -177,15 +177,17 @@ public void testHeader(){
 
         ResultSet queryResult = api.query("social", "MATCH (a:person)-[r:knows]->(b:person) RETURN a,r, a.age");
 
-        Assert.assertNotNull(queryResult.getHeader());
         Header header = queryResult.getHeader();
-
+        Assert.assertNotNull(header);
+        Assert.assertEquals("HeaderImpl{"
+            + "schemaTypes=[COLUMN_SCALAR, COLUMN_SCALAR, COLUMN_SCALAR], "
+            + "schemaNames=[a, r, a.age]}", header.toString());
+//        Assert.assertEquals(-1901778507, header.hashCode());
```

Comment:
@DvirDukhan something is strange about the HeaderImpl.hashCode(), it's not deterministic, on each run the value is different despite the fact that the values are the same